source/material_model/rheology/friction_options.cc

```diff
@@ -0,0 +1,849 @@
+
+                  rate_and_state_parameter_a = rate_and_state_parameter_a + slope_s_for_a
+                                               * log10((ref_v_for_a + current_edot_ii * cellsize)/ref_v_for_a);
+
+                  const double dependent_critical_slip_distance = critical_slip_distance + slope_s_for_L
+                                                                  * log10((ref_v_for_L + current_edot_ii * cellsize)/ref_v_for_L);
+
+                  // Calculate effective friction according to Equation (4) in Sobolev and Muldashev (2017):
+                  // mu = mu_st + a ln(V/V_st) + b ln((theta Vst)/L)
+                  // Effective friction is calculated by multiplying the friction coefficient with the
+                  // effective_friction_factor to account for effects of pore fluid pressure:
+                  // mu = mu(1-p_f/sigma_n) = mu*, with (1-p_f/sigma_n) = 0.03 for subduction zones.
+                  // Their equation is for friction coefficient, while ASPECT takes friction angle in radians,
+                  // so conversion with tan/atan().
+                  current_friction = atan(effective_friction_factor[j]
+                                          * (tan(current_friction)
+                                             + rate_and_state_parameter_a
+                                             * log((current_edot_ii * cellsize ) / quasi_static_strain_rate)
+                                             + rate_and_state_parameter_b
+                                             * log((theta * quasi_static_strain_rate ) / dependent_critical_slip_distance)));
+                }
+              break;
+            }
+            case regularized_rate_and_state_friction:
+            {
+              // TODO as the cell size is used to compute the slip velocity as cell_size * strain_rate,
+              //  come up with a better representation of the slip length.
+              double cellsize = 1.;
+              if (current_cell.state() == IteratorState::valid)
+                {
+                  cellsize = current_cell->extent_in_direction(0);
+                  // Calculate the state variable theta
+                  // theta_old loads theta from previous time step
+                  const double theta_old = composition[theta_composition_index];
+
+                  // Get the values for a and b
+                  const double rate_and_state_parameter_a = calculate_depth_dependent_a_and_b(position,j).first;
+                  const double rate_and_state_parameter_b = calculate_depth_dependent_a_and_b(position,j).second;
+                  const double critical_slip_distance = get_critical_slip_distance(position,j);
+
+                  // Equation (7) from Sobolev and Muldashev (2017)
+                  const double theta = compute_theta(theta_old, current_edot_ii, cellsize, critical_slip_distance);
+
+                  // as RSF parameter a is used to divide current_friction will be nan without this if
```

Comment:
yes, I need this case, because else all friction angles outside the fault will be nan. As currently I limit rate-and-state behaviour to the fault by setting a and b zero for all other materials. Which results in not modifying the friction angle for the classical log-formulation of RSF